bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -3008,6 +2976,251 @@ private static boolean containsType(Set<Object> valueSpace, Type type) {
         return true;
     }
 
+    /**
+     * Check if left hand side ordered type value is less than right hand side ordered type value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side ordered type value is less than right hand side ordered type value, else false.
+     */
+    public static boolean compareValueLessThan(Object lhsValue, Object rhsValue) {
```

Comment:
Shall we move these value compare methods to a different class? IMO these are not part of the Type checker

Reply:
comments addressed via 3515735 & [`a648baa`](https://github.com/ballerina-platform/ballerina-lang/pull/29064/commits/a648baa131843d3f1dcd88cb12640c6acab3e801)